### PR TITLE
add fread support for text= single line without \r or \n

### LINKED ISSUE #4689: fread with single line text
When calling fread with the text="content,to,parse" the behaviour is not as expected if the value is a string without a newline.
In that case, the value of `text=` is interpreted as a file name, the same behaviour of the `input=` parameter.

Example:
```
> fread(text = "a,b,c")
Error in fread(text = "a,b,c") : file not found: a,b,c 
```
I expected a data.table with three columns (a, b, c) and no rows, or a data.table with one row and columns V1, V2, V3, depending on the value of `col.names`.

```
R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6
```

### PR DESCRIPTION
**Pull request by ben-schwen**
Closes #4689. `fread(text = "a b c")` works now and produces an `data.table` with 0 rows and 3 cols. Beforehand this resulted in an error because single line text without `\r` or `\n` was treated like `fread(input = text)`.

### CODE REVIEW

File: R/fread.R [hunk 1]
```diff
@@ -41,6 +41,8 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir(), tz="")
     } else {
       # avoid creating a tempfile() for single strings, which can be done a lot; e.g. in the test suite.
       input = text
+      if (!length(grep("\\n|\\r", input)))
```
Comment (jangorecki) [inline R/fread.R:44]: I would link the issue in comment so it will be easier to understand that code in future

**Comment by torbjorn** [comment]
Is this really fixed? I'm seeing this with 1.14.1:


        > packageVersion("data.table")
        [1] ‘1.14.2’
        > fread(text="a,b,c\n", header=F)
           V1 V2 V3
        1:  a  b  c
        > fread(text="a,b,c", header=F)
        Error in fread(text = "a,b,c", header = F) : file not found: a,b,c
        >

**Comment by mattdowle** [comment]
@ben-schwen @torbjorn once 1.14.4 is accepted on CRAN, I'll tag that in the patch-14.4 branch and then probably move forward with backporting bug fixes like this to 1.14.6 on that branch together with new features in dev that are stable and shouldn't have any impact revdeps; e.g. the new `env=` argument.

**Comment by ben-schwen** [comment]
@mattdowle It's true once I click on the milestones or check the milestone below it says 1.14.5. In the timeline of commits, mentions and merges it is displayed as 1.14.1 (see following image) 
Screenshot_20221015_214158

**Comment by mattdowle** [comment]
Thanks. Yes it's the same milestone that's since been renamed. Both the closed issue and the closed PR are marked 1.14.5 on the right though. Maybe in future then we should have a 'dev' or 'unstable' milestone and then move items to the release branch. Like R-devel does it.

**Comment by torbjorn** [comment]
It's 1.14.8 now and this seems not to be fixed:

    > fread( text="hi,ho,silver", header=FALSE )
    Error in fread(text = "hi,ho,silver", header = FALSE) : 
      file not found: hi,ho,silver

with a newline:

    > fread( text="hi,ho,silver\n", header=FALSE )
       V1 V2     V3
    1: hi ho silver

version:

    > packageVersion("data.table")
    [1] ‘1.14.8’

**Comment by ben-schwen** [comment]
@torbjorn this is fixed in the current dev version. `1.14.2 - 1.14.8` were only small updates fixing things to keep it on CRAN, accommodate `R dev version`, etc.

**Pull request merged**